### ISSUE TITLE
Add support for ctrl+k

### ISSUE THREAD

**Issue by andrewpareles** [voideditor/void#11]
When the user presses ctrl+k, an input box should appear inline with the code that they were selecting. This is somewhat difficult to do because the vscode extension API alone doesn't support it - it requires creating a new component in the IDE. 

We think we can modify vscode's built-in "codelens" or "zone widget" components to build the input box to ctrl+K, but we are open to alternatives like taking code from the "diffEditor".

![inline4](https://github.com/user-attachments/assets/22f147d4-72d2-4aaf-9d69-caf3c4a3a10f)

**Comment by AllyourBaseBelongToUs** [comment]
Do everything Cursor does but OSS?

Guess you guys are on the right track

**Comment by andrewpareles** [comment]
We don’t plan on just copying Cursor :) 

Inline edits are a pretty core feature that most AI extensions / editors support.

**Comment by sunmorgan** [comment]
Working on it, should open a pr soon

**Comment by ben-localhost** [comment]
does commenting count as contribution ? can i put that on my resume ?

**Comment by imcheeseberg** [comment]
This Feature is available on VSCode Insider Version, called 'editorInsets', you can test it by an extension which has add   "enabledApiProposals": ["editorInsets"] to package.json，but the feature is not completed as no 'setHeight' is provided.

Here is a issue lasted 5 years long.... https://github.com/microsoft/vscode/issues/85682

**Comment by andrewpareles** [comment]
> This Feature is available on VSCode Insider Version, called 'editorInsets', you can test it by an extension which has add "enabledApiProposals": ["editorInsets"] to package.json，but the feature is not completed as no 'setHeight' is provided.
> 
> Here is a issue lasted 5 years long.... [microsoft/vscode#85682](https://github.com/microsoft/vscode/issues/85682)

If this is true this is a great insight. Will look into this soon - thanks for the suggestion.

**Comment by andrewpareles** [comment]
> This Feature is available on VSCode Insider Version, called 'editorInsets', you can test it by an extension which has add "enabledApiProposals": ["editorInsets"] to package.json，but the feature is not completed as no 'setHeight' is provided.
> 
> Here is a issue lasted 5 years long.... [microsoft/vscode#85682](https://github.com/microsoft/vscode/issues/85682)

Great insight - thanks for this!

**Comment by qiao-wei** [comment]
why do I pressed the "ctrl + k" , show the error "command 'void.ctrl+k' not found"

**Comment by qiao-wei** [comment]
why does it show the error "command 'void.ctrl+k' not found" shen i press the "ctrl+k"

**Comment by andrewpareles** [comment]
Ctrl+K still hasn't been merged into the main branch.

**Comment by andrewpareles** [comment]
@imcheeseberg Any idea how to get editorInsets up and running?

**Comment by andrewpareles** [comment]
For reference, see https://github.com/microsoft/vscode/pull/66418 and https://github.com/microsoft/vscode/issues/85682. We're going with editorInsets, the modern version of these. #119 has a Hello World editor inset.

**Comment by sunmorgan** [comment]
> For reference, see [microsoft/vscode#66418](https://github.com/microsoft/vscode/pull/66418) and [microsoft/vscode#85682](https://github.com/microsoft/vscode/issues/85682). We're going with editorInsets, the modern version of these. #119 has a Hello World editor inset.

I don't think you can actually use it though right, they specifically mentioned that you can't use it in production.
"/**
 * This is the place for API experiments and proposals.
 * These API are NOT stable and subject to change. They are only available in the Insiders
 * distribution and CANNOT be used in published extensions.
 *
 * To test these API in local environment:
 * - Use Insiders release of VS Code.
 * - Add `"enableProposedApi": true` to your package.json.
 * - Copy this file to your project.
 */"

**Comment by andrewpareles** [comment]
This is true if you're just building an extension. We're not just building an extension, we're building a full IDE.

**Comment by andrewpareles** [comment]
The basic layout for Ctrl+K is now on main. It'll be implemented soon!

**Comment by manisenkov** [comment]
@andrewpareles is it possible to change those keybindings?

Cmd+K is a standard shortcut for clearing the terminal window (it is also default in VS Code), but in Void when I press it opens the inline completion in the current tab instead.

**Comment by andrewpareles** [comment]
Yes, you can change them in Void Settings > Keyboard Settings! Just find the Cmd+K

**Comment by manisenkov** [comment]
Thank you! I actually overlooked the command in the keyboard settings screen. It's called "**Void: Quick Edit**" and doesn't have any conditions. I set it to !terminalFocus, and now it works as expected. 👍
